docs/components/codeblock.md

```diff
@@ -0,0 +1,268 @@
+# CodeBlock
+
+**KCodeBlock** is a code block component with built-in search and filter functionalities.
+
+<KCodeBlock
+  id="code-block-default"
+  :code="code"
+  language="json"
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-default"
+  :code="code"
+  language="json"
+/>
+```
+
+## Props
+
+### id
+
+* **Type**: `string`
+* **Required**: no
+* **Default**: `'code-block'`
+
+An ID used to identify form elements like the search input and its label.
+
+### code
+
+* **Type**: `string`
+* **Required**: yes
+
+The code that will be rendered as a text node inside of a `<pre><code></code></pre>` fragment.
+
+### language
+
+* **Type**: `string`
+* **Required**: yes
+
+The syntax language of `props.code` (e.g. `'json'`).
+
+### isSearchable
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `false`
+
+Shows an actions bar with a search input and related action buttons.
+
+<KCodeBlock
+  id="code-block-is-searchable"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-is-searchable"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+```
+
+### isProcessing
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `false`
+
+Allows controlling the processing state from outside the component. This allows a parent component to show the processing icon when it’s, for example, currently syntax highlighting the code.
+
+<KCodeBlock
+  id="code-block-is-processing"
+  :code="code"
+  language="json"
+  is-searchable
+  is-processing
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-is-processing"
+  :code="code"
+  language="json"
+  is-processing
+/>
+```
+
+### query
+
+* **Type**: `string`
+* **Required**: no
+* **Default**: `''`
+
+Used as the initial value of the search input. Can be used to initialize a code block with a query which was read from client storage.
+
+<KCodeBlock
+  id="code-block-query"
+  :code="code"
+  language="json"
+  is-searchable
+  query="key"
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-query"
+  :code="code"
+  language="json"
+  is-searchable
+  query="key"
+/>
+```
+
+### showCopyButton
```

Comment:
nit: Can we rename this prop to just `copy` ? We're trying to get away from prefixed props, e.g. `is*`, `has*`, `show*`

Reply:
> We're trying to get away from prefixed props, e.g. is*, has*, show*

I’m curious to hear a rational for this as I don’t agree with it. Names like `copy`, especially when they’re (also) verbs, are very ambiguous and it’s nice to know when something is a boolean. IDE support for extracting prop type information is generally poor (though this is drastically improving with the advent of composition API and script setup authored components) so seeing at a glance that a prop `is-something` is expected allows you to infer that you’re to provide a boolean. When I face a prop `copy`, my first intuition would be to provide my own copy handler which would be incorrect in this case.

I really don’t want to change this.

Reply:
Then as an alternative I'd suggest `copyButton` as the prop name. `true` indicates it's showing, set to `false` to hide.